### PR TITLE
[flutter_tools] Add some useful commands to the README.md

### PR DESCRIPTION
**Pull request by zanderso**
## Description

Adds instructions for running, analyzing, and testing to the flutter_tools README.md

## Related Issues

https://github.com/flutter/flutter/issues/5548

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (Please read [Handling breaking changes]). *Replace this with a link to the e-mail where you asked for input on this proposed change.*
- [x] No, this is *not* a breaking change.

### CODE REVIEW

File: packages/flutter_tools/README.md [hunk 1]
```diff
@@ -1,13 +1,36 @@
...
-To run the tests, ensure that no devices are connected,
-then navigate to `flutter_tools` and execute:
+Be sure to follow the instructions on [CONTRIBUTING.md](../../CONTRIBUTING.md)
+to set up your development environment.
 
+### Running
+
+To run Flutter Tools from source, in this directory run:
+```shell
+$ ../../bin/cache/dart-sdk/bin/dart bin/flutter_tools.dart
```
Comment (jonahwilliams) [inline packages/flutter_tools/README.md]: Add a note that this requires the dart_sdk to have been downloaded first from a `flutter` command.

File: packages/flutter_tools/README.md [hunk 2]
```diff
@@ -1,13 +1,36 @@
...
+```
+followed by command line arguments, as usual.
+
+### Analyzing
+
+To run the analyzer on Flutter Tools, in this directory run:
+```shell
+$ ../../bin/cache/dart-sdk/bin/dartanalyzer .
```
Comment (jonahwilliams) [inline packages/flutter_tools/README.md]: Flutter analyze should work as well

File: packages/flutter_tools/README.md [hunk 3]
```diff
@@ -1,13 +1,36 @@
...
+```
+
+### Testing
+
+To run the tests in the `test/` directory, first ensure that there are no
```
Comment (jonahwilliams) [inline packages/flutter_tools/README.md]: To run all tests precompiled, first set `FLUTTER_ROOT` to an appropriate value. then run `../../bin/cache/dart-sdk/bin/pub run build_runner test`

Comment (jonahwilliams) [reply packages/flutter_tools/README.md]: Maybe also a note that the integration/command shard is really slow currently

**Review by christopherfujino — APPROVED** [review]
LGTM

File: packages/flutter_tools/README.md (hunk 1 above)
Comment (zanderso) [reply packages/flutter_tools/README.md]: Added a "Setup" section

File: packages/flutter_tools/README.md (hunk 2 above)
Comment (zanderso) [reply packages/flutter_tools/README.md]: Switched to that since it's shorter. Thanks!

File: packages/flutter_tools/README.md (hunk 3 above)
Comment (zanderso) [reply packages/flutter_tools/README.md]: Added.

File: packages/flutter_tools/README.md [hunk 4]
```diff
@@ -1,13 +1,58 @@
...
+
+First, ensure that the Dart SDK and other necessary artifacts are available by
+invoking the Flutter Tools wrapper script. In this directory run:
+```shell
+$ ../../bin/flutter --version
+```
+
+### Running
+
+To run Flutter Tools from source, in this directory run:
```
Comment (jonahwilliams) [inline packages/flutter_tools/README.md]: Sorry, one more:

To re-generate the snapshotted tool, delete `bin/cache/flutter_tools.stamp` and `bin/cache/flutter_tools.snapshot`

**Review by jonahwilliams — APPROVED** [review]
LGTM with comment

File: packages/flutter_tools/README.md (hunk 4 above)
Comment (zanderso) [reply packages/flutter_tools/README.md]: Added.

**Pull request merged**